fix(pkg): Add User-Agent to headers

lasjorg opened the pull request:
This PR should (potentially) fix an issue with Glitch we are seeing where the request function gets a 403 and is redirected to a page with the title Maintenance. This causes the res.json to send HTML instead of JSON.

I have not yet had anybody else give confirmation of this fix because Glitch is really unstable right now. I would just like to get this PR made now anyway. Also, I just chose some random user-agent string, not sure what the best option is.

The diff is a little noisy because of the prettier formatting, the PR change is on lines 26-29 in index.js

Closes #2, closes #3, and closes #4

More info here:
https://github.com/freeCodeCamp/freeCodeCamp/issues/39095

Here is a starter project with the change added to the dependencies (dependency link is to my own repo)
https://glitch.com/edit/#!/basic-node-and-express-challenges-with-new-pkg

lasjorg committed:
fix(pkg): Add User-Agent to headers

RandellDawson commented:
@lasjorg Should this fix issue #3 and #4 in this repo?

lasjorg commented:
Technically yes, but those issues are listing reasons that are not actually breaking the challenge (#3 next missing from the parameters, #4 bowser version < 2 on the redirect page).

I can add them to the PR closes issues if you want?

Sky020 commented:
Hey, lasjorg. How did you choose/find that specific user agent?

lasjorg commented:
@Sky020 It's just some random one I found in the examples of this npm package.
https://www.npmjs.com/package/user-agents

I initially used Postman's user-agent and when that worked I just picked a more common one.

ojeytonwilliams commented:
@lasjorg Glitch is still having issues talking to its api, so I can't confirm this yet.  

As for the noise, don't worry about it for this PR, but you can split the formatting into a separate commit, so that reviewers can ignore it.

lasjorg commented:
We are getting a lot of forum posts about this issue. Can nobody comfirm this fix? I'd like to get this merged so we can see if it helps or not.

@RandellDawson @Sky020 @ojeytonwilliams

Here is the link for the .env challenge:
https://www.freecodecamp.org/learn/apis-and-microservices/basic-node-and-express/use-the-.env-file

Here is the project with the fix from this PR in the dependency. Just paste it in the for the solution, it should pass (the project has to wake up first).
https://basic-node-and-express-challenges-with-new-pkg.glitch.me

And just for reference here is a project that has the exact same code for the solution, but without the PR fix, it should fail.
https://lasjorg-fcc-node-and-express.glitch.me/

I also updated the OP with the link to the project in editor mode. I added the normal full page link by mistake.

Sky020 commented:
Sorry about the delay. You are right - we should get this moving, as many are encountering the issue.

From what I have been able to see, your solution does solve the issue. I am interested in why this is/all-of-a-sudden became an issue with Glitch, though...

ShaunSHamilton commented on index.js:
```diff
@@ -14,53 +13,65 @@ var globals = require('./globals');
+      } catch (e) {
         console.log(e);
         process.env.MESSAGE_STYLE = msgStyle;
         next(e);
```
I am not entirely familiar with the working of this function, but as mentioned in another issue, `next()` either needs to be passed to be used here, or this line needs to be removed.

lasjorg replied:
I guess it's fine to fix it in this PR. Done.

ojeytonwilliams replied:
It's fine, I just won't squash the commits.

ojeytonwilliams replied:
So, this is kinda weird, because it seems to be passing around stuff that's already available via closure.  I think it's fine, just confusingly written.

ShaunSHamilton reviewed (changes requested):
Once, this gets resolved, I would approve.

lasjorg committed:
add: missing next parameter

ShaunSHamilton reviewed (approved):
LGTM 👍

lasjorg commented:
> I am interested in why this is/all-of-a-sudden became an issue with Glitch, though...

I can't say for sure but the blocking of [pinging services](https://support.glitch.com/t/a-talk-about-the-new-ping-services/27194) might somehow [be related](https://support.glitch.com/t/glitch-app-to-glitch-app-403s/26551). It should be temporary as far as I can tell from reading the forum a bit, but there is no ETA on when they will unblock.

lasjorg commented:
Can someone please merge this PR?

ojeytonwilliams commented:
Really nice job @lasjorg.  I was finally able to test it out and it worked beautifully.

The change *should* be live now, but there may be some caching issues, so I'll try again later.

lasjorg commented:
@ojeytonwilliams Thank you for the merge. It works for me. I did have to remove the dependency and add it back to get the new version.

---

Side note: I forgot that I wanted to remove the `engines` property from package.json. I don't think it makes any sense to have it as Glitch just ignores it anyway (terminal message: Could not find node 4.4.5, using 10) and it's such an old node version. I'm not sure if it really matters, but we might as well remove the property. I might make a new PR at some point just to get it done.
